src/mito2/src/memtable.rs

```diff
@@ -191,14 +227,13 @@ pub trait Memtable: Send + Sync + fmt::Debug {
 
     /// Returns the ranges in the memtable.
     ///
-    /// The `for_flush` flag is true if the flush job calls this method for flush.
     /// The returned map contains the range id and the range after applying the predicate.
     fn ranges(
         &self,
         projection: Option<&[ColumnId]>,
         predicate: PredicateGroup,
         sequence: Option<SequenceRange>,
-        for_flush: bool,
+        options: RangesOptions,
```

Comment:
Can we merge `projection`/`predicate`/`sequence` into `RangeOptions`?

Reply:
Adding the projection will introduce a lifetime parameter to the options which is I'd like to avoid. I'm considering moving predicate and sequence to the options.